tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -109,61 +210,122 @@ spec:
             fi
 
             for manifest_digest in $manifest_digests; do
-              for tag in ${TAGS}; do
-                # Iterate over both rh-registry-repo and registry-access-repo
-                for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
-                  echo "Creating InternalRequest to sign image with tag ${tag}:"
-                  echo "- reference=${registry_reference}:${tag}"
-                  echo "- manifest_digest=${manifest_digest}"
-                  echo "- requester=$(params.requester)"
-
-                  internal-request -r "${request}" \
-                      -p pipeline_image=${pipeline_image} \
-                      -p reference=${registry_reference}:${tag} \
-                      -p manifest_digest=${manifest_digest} \
-                      -p requester=$(params.requester) \
-                      -p config_map_name=${config_map_name} \
-                      -l ${TASK_LABEL}=${TASK_ID} \
-                      -l ${PIPELINERUN_LABEL}=$(params.pipelineRunUid) \
-                      -s false
-                  ((++count))
-
-                  if [ "$count" -eq "$N" ]; then
-                      wait-for-ir -l ${TASK_LABEL}=${TASK_ID} -t $(params.requestTimeout)
-                      count=0
+
+              set -x
+              find_signatures --pyxis-graphql-api "${PYXIS_GRAPHQL_URL}" \
+                  --manifest_digest "${manifest_digest}" \
+                  --repository "${repository}" \
+                  --output_file "/tmp/${manifest_digest}"
+              set +x
+
+              # Iterate over both rh-registry-repo and registry-access-repo
+              for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
+
+                for tag in ${TAGS}; do
+
+                  # is tag signed?
+                  set +e
```

Comment:
`set +e` is about errors no? Is there another way that is would not require the `set +e`?

Reply:
> set +e is about errors no?

Right, sorry.

I would just do it directly in the `if` like I also do in [another PR](https://github.com/konflux-ci/release-service-catalog/pull/601/files#diff-7e331e2b56eb82782bd8f5a0ce886f80d2cb038643e283f18eef69fcd8778e0fR81).

```sh
if ! grep -q "^${registry_reference}:${tag}$" "/tmp/${manifest_digest}" ; then
 sign
else
  skip
fi
```

And it would probably make sense to also switch the order:
```sh
if grep -q "^${registry_reference}:${tag}$" "/tmp/${manifest_digest}" ; then
 skip
else
  sign
fi
```

Reply:
I prefer the 1st suggestion. fixed